### ISSUE TITLE
Ratpack Compatability

### ISSUE THREAD

**Issue by adamdougal** [ahus1/prometheus-hystrix#14]
Hi, I'm having an issue integrating this with a Ratpack application. I'm getting the following error on startup:

```
INFO: An exception was caught and reported. Message: java.lang.IllegalStateException: Cannot install Hystrix integration because another concurrency strategy (class com.netflix.hystrix.strategy.concurrency.HystrixConcurrencyStrategyDefault) is already installed
java.lang.IllegalStateException: Cannot install Hystrix integration because another concurrency strategy (class com.netflix.hystrix.strategy.concurrency.HystrixConcurrencyStrategyDefault) is already installed
```

It looks like this bit of `prometheus-hystrix` code:
https://github.com/ahus1/prometheus-hystrix/blob/master/src/main/java/com/soundcloud/prometheus/hystrix/HystrixPrometheusMetricsPublisher.java#L94

is clashing with this bit of Ratpack code:
https://github.com/ratpack/ratpack/blob/master/ratpack-hystrix/src/main/java/ratpack/hystrix/HystrixModule.java#L90

Is it necessary to register the default `HystrixConcurrencyStrategy` in `prometheus-hystrix` or can it be removed?

**Comment by ahus1** [comment]
The current prometheus-hystrix code tries to re-register any previously registered modules. But I think I can optimize it in a way that I don't trigger the default implementations where there is no previously registered module. 

For now, you could try to re-order the initialization: it should work if you first initialize the ratpack-hystrix, and afterwards prometheus-hystrix.

Please let me know if this workaround works for you.

**Comment by adamdougal** [comment]
@ahus1 this seems to have fixed the issue!

Thanks for working on a fix so quickly 👍

**Comment by ahus1** [comment]
This has been released as version 3.4.0 and should be available on Maven Central soon. There have been changes to also support JDK6 and JDK7 in addition to JDK8 (as both are also supported by Hystrix and Prometheus Java Client).
